From 91daa03935095122c20dc28ba466574772294803 Mon Sep 17 00:00:00 2001
From: Xuefu Zhang <xuefu@apache.org>
Date: Fri, 27 Mar 2015 16:19:48 +0000
Subject: [PATCH 0111/1164] HIVE-10087: Beeline's --silent option should
 suppress query from being echoed when running
 with -f option (Naveen via Xuefu)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1669615 13f79535-47bb-0310-9956-ffa450edef68
---
 .../src/java/org/apache/hive/beeline/BeeLine.java  |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLine.java b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
index c60c11d..e7b4ed4 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLine.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLine.java
@@ -799,10 +799,14 @@ private int executeFile(String fileName) {
   }
 
   private int execute(ConsoleReader reader, boolean exitOnError) {
+    String line;
     while (!exit) {
       try {
         // Execute one instruction; terminate on executing a script if there is an error
-        if (!dispatch(reader.readLine(getPrompt())) && exitOnError) {
+        // in silent mode, prevent the query and prompt being echoed back to terminal
+        line = getOpts().isSilent() ? reader.readLine(null, ConsoleReader.NULL_MASK) : reader.readLine(getPrompt());
+
+        if (!dispatch(line) && exitOnError) {
           return ERRNO_OTHER;
         }
       } catch (Throwable t) {
-- 
1.7.9.5

